exasol_advanced_analytics_framework/event_handler/event_handler_base.py

```diff
@@ -0,0 +1,19 @@
+from abc import ABC, abstractmethod
+from exasol_advanced_analytics_framework.event_handler.event_handler_result import \
+    EventHandlerResult
+
+
+class EventHandlerBase(ABC):
+    def __init__(self):
+        self.state = None
+
+    @abstractmethod
+    def handle_event(self) -> EventHandlerResult:
```

Comment:
here are some parameters missing, such as RowIterator and Context

Reply:
Rowiterator, you mean the iterator from exa ctx, right?

Reply:
Yes, but we need to abstract it, in case we want to run the event handler somewhere else. Or, we keep the interface and build wrapper when we don't use UDFs for the event handler. However, you might need to filter out certain columns which are important for the framework, but the use shouldn't have access, too. I implemented something like that in the data-science-utils-python